tests/test_reusable_executor.py

```diff
@@ -433,6 +433,22 @@ def test_reusable_executor_resize(self):
         assert len(executor._processes) == 2
         assert old_pid in list(executor._processes.keys())
 
+    @pytest.mark.parametrize("kill_workers", [True, False])
+    def test_reusable_executor_resize_many_times(self, kill_workers):
+        # Tentative non-regression test for a deadlock when shutting down
+        # the workers of an executor prior to resizing it.
+        kwargs = {'timeout': None, 'kill_workers': kill_workers}
+        executor = get_reusable_executor(max_workers=2, **kwargs)
+        executor.map(id, range(2))
+        sizes = [12, 2, 1, 12, 6, 1, 8, 5]
+        with warnings.catch_warnings(record=True):
+            # Cause all warnings to always be triggered.
+            warnings.simplefilter("always")
+            for size in sizes:
+                executor = get_reusable_executor(max_workers=size, **kwargs)
```

Comment:
I think you need to use `reuse=False` to make sure the executor is shutdown.
In this test, you are resizing the executor (spawning more processes with `_adjust_process_count`)  and I am not sure this is what you want to test.

Reply:
Indeed. I will add a new parameter.